src/app/newsletter/newletter-page-client.tsx

```diff
@@ -0,0 +1,88 @@
+"use client";
+
+import React, { Suspense } from "react";
+import Link from "next/link";
+import { ArrowLeftIcon } from "@radix-ui/react-icons";
+import { NewsletterData } from "@/helpers/types";
+import { usePaginatedResult } from "../components/client/hooks/use-pagination";
+import NewsletterCard from "../components/client/newsletter-card";
+import Pagination from "../components/client/Pagination";
+
+const NewsletterPageClient = ({ newsletters }: { newsletters: NewsletterData[] }) => {
+  const [sortKey, setSortKey] = React.useState<"newest" | "oldest">("newest");
+
+  const memoizedNewsletters = React.useMemo(() => {
+    switch (sortKey) {
+      case "newest":
+        return newsletters.sort((a, b) => b.issueNumber - a.issueNumber);
+      case "oldest":
+        return newsletters.sort((a, b) => a.issueNumber - b.issueNumber);
+      default:
+        return newsletters;
+    }
+  }, [newsletters, sortKey]);
+
+  const { currentPage, paginatedResult, setCurrentPage } = usePaginatedResult(memoizedNewsletters, 9);
+  const pageSize = 9;
```

Comment:
pageSize can be defined earlier and passed to the usePaginatedResult hook. So there is one source of truth